File: April 2022/April_23.cpp
class Solution {
public:
    vector<string> vec;
    int val = 0;
    // Encodes a URL to a shortened URL.
    string encode(string longUrl) {
        vec.push_back(longUrl);
        return "http://tinyurl.com/" + to_string(val++);
    }

    // Decodes a shortened URL to its original URL.
    string decode(string shortUrl) {
        return vec[stoi(shortUrl.substr(19))];
    }
};

// Your Solution object will be instantiated and called as such:
// Solution solution;
// solution.decode(solution.encode(url));

File: April 2022/April_23.java
public class Codec {
    List<String> list = new ArrayList<>();
    int val = 0;
    // Encodes a URL to a shortened URL.
    public String encode(String longUrl) {
        list.add(longUrl);
        return "http://tinyurl.com/" + val++;
    }

    // Decodes a shortened URL to its original URL.
    public String decode(String shortUrl) {
        int val = Integer.parseInt(shortUrl.substring(19));
        return list.get(val);
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(url));
